airflow/jobs/base_job.py

```diff
@@ -117,7 +117,7 @@ def most_recent_job(cls, session=None) -> Optional['BaseJob']:
         :param session: Database session
         :rtype: BaseJob or None
         """
-        return session.query(cls).order_by(cls.latest_heartbeat.desc()).limit(1).first()
+        return session.query(cls).filter(cls.hostname == self.hostname).order_by(cls.latest_heartbeat.desc()).limit(1).first()
```

Comment:
This change will prevent our API from returning the correct state.
https://github.com/apache/airflow/blob/b007fc33d481f0f1341d1e1e4cba719a5fe6580d/airflow/www/views.py#L418
https://github.com/apache/airflow/blob/349b0811c3022605426ba57d30936240a7c2848a/airflow/api_connexion/endpoints/health_endpoint.py#L30